corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/config/storage/localstorage.js

```diff
@@ -0,0 +1,53 @@
+/*global FormplayerFrontend */
+
+FormplayerFrontend.module("Entities", function (Entities, FormplayerFrontend, Backbone, Marionette, $, _) {
+    var findStorageKey = function (entity) {
+        // use a model's urlRoot value
+        if (entity.urlRoot) {
+            return _.result(entity, "urlRoot");
+        }
+        // use a collection's url value
+        if (entity.url) {
+            return _.result(entity, "url");
+        }
+
+        throw new Error("Unable to determine storage key");
+    };
+
+    var storageCache = {};
+    var getStorage = function (key) {
+        var storage = storageCache[key];
+        if (storage) {
+            return storage;
+        }
+        var newStorage = new Backbone.LocalStorage(key);
+        storageCache[key] = newStorage;
+        return newStorage;
+    };
+
+    var StorageMixin = function (entityPrototype) {
+        var storageKey = findStorageKey(entityPrototype);
+        return {localStorage: getStorage(storageKey)};
+    };
+
+    var getEntity = function (constructorString) {
+        var bits = constructorString.split("."),
```

Comment:
This is actually from https://github.com/davidsulc/marionette-gentle-introduction/commit/53f0f05b95c440e376ef74a60b4c41bbf7227de8

Happy to refactor but we should discuss first whether this localStorage workflow is actually correct (basically we just need some way to access the list of apps, might be that a queriable HQ API would be better